### DISCUSSION TITLE
Failed to acquire connection from pool

### DISCUSSION THREAD

**Discussion by efrain2007** [SeaQL/sea-orm#1521] [Q&A]
Failed to acquire pool connection in version 0.11 when making more than 50 queries

**Comment by tyt2y3** [comment]
This is expected current behaviour. This happens when the connection pool is fully-utilized.
You can increase the pool size, or better, look through you code into why it consumes many connections for too long. A transaction exclusively holds the connection, so it is one thing to look for.
If you find yourself doing too many transactions concurrently, you should queue them and do them one by one, which as a side-effect will also make the database less busy.

I have created an issue here https://github.com/SeaQL/sea-orm/issues/1523 for possible improvements.

**Reply by yuhaya** [reply]
hi，Failed to Get Connection from pool is often seen, but there is no record in the mysql slow log, I just run it in a local test environment, there is not a very large amount of data, but this problem still occurs frequently.
Try setting the acquire_timeout adjustment to 90 seconds, very large, but the problem still occurs frequently
I guess there may be some deeper reasons.

**Reply by efrain2007** [reply]
I managed to solve it so that this error no longer happens

**Reply by tyt2y3** [reply]
@efrain2007 It would be nice if you can share your diagnosis and solution, it might be relevant to others.

**Comment by efrain2007** [comment]
![image](https://user-images.githubusercontent.com/65697999/223792635-55cf7ef9-a6f4-43b9-953f-a58fd51eacd3.png)
That is what I wanted to keep the connections at 0, so not when you want to acquire the connection and when you find it closed, what you will do is open a new one.

**Reply by yuhaya** [reply]
thanks, i'll try

**Reply by yuhaya** [reply]
Although this error does not appear, but the sql execution is often blocked all the time

**Reply by yuhaya** [reply]
It may be that the Linux file descriptor setting is too small. I solved this problem by adjusting this value.
